project/src/main/java/com/google/sps/servlets/InvestmentCalculator.java

```diff
@@ -0,0 +1,131 @@
+// Copyright 2021 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package com.google.sps.servlets;
+
+import com.google.gson.Gson;
+import com.google.sps.data.*;
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.Date;
+import java.util.List;
+
+import java.sql.PreparedStatement;
+import java.sql.ResultSet;
+import java.sql.Connection;
+import java.sql.SQLException;
+import javax.sql.DataSource;
+
+import java.util.logging.Level;
+import java.util.logging.Logger;
+import com.google.cloud.datastore.StructuredQuery.PropertyFilter;
+import com.google.cloud.datastore.*;
+
+import java.util.Calendar; 
+import java.util.TimeZone;
+
+
+public class InvestmentCalculator {
+    private static final Logger LOGGER = Logger.getLogger(AuthServlet.class.getName());
+    public int calculateNetWorth(Connection conn, int userId, int competitionId) throws SQLException{
```

Comment:
Perhaps `Connection conn` should be an instance variable of `InvestmentCalculator` to avoid being passed around so much.
@leonnemets WDTY?

Reply:
I really only did it this way to avoid having to error catch the connection multiple times. I have no idea if that is good or bad, I chose it because lazy.

Reply:
@texm  probably agree with you. To avoid try/catch everywehere you can declare that a function throws, eg throws SqlException.